src/clarity-angular/datagrid/datagrid-hideable-column.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2016 -2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { TemplateRef } from "@angular/core";
+import { Subject } from "rxjs/Subject";
+import { Observable } from "rxjs/Observable";
+
+
+/**
+ * @class DatagridHideableColumn
+ *
+ * @description
+ * A utility class for that adds hide/show functionality to a column, its cells and enables a toggler in the
+ * DatagridColumnToggle Component.
+ *
+ */
+export class DatagridHideableColumn {
+
+    /**
+     * @property hiddenChanges
+     *
+     * @description
+     * A stream of state changes an instance of DatagridHideableColumn will broadcast to subscribers.
+     *
+     * @type {Subject<boolean>}
+     */
+    private hiddenChanges: Subject<boolean> = new Subject<boolean>();
```

Comment:
`hiddenChanges` suggests to me that changes were hidden, but I believe it's a property that notifies the state of hideable columns. But It shouldn't block this PR at all as this is so minor nit.